src/analysis/bounded_variation.lean

```diff
@@ -634,6 +634,148 @@ lemma monotone_on.has_locally_bounded_variation_on {f : α → ℝ} {s : set α}
   has_locally_bounded_variation_on f s :=
 λ a b as bs, ((hf.evariation_on_le as bs).trans_lt ennreal.of_real_lt_top).ne
 
+
+
+namespace has_locally_bounded_variation_on
+
+variables {f : α → E} {s : set α} (hf : has_locally_bounded_variation_on f s)
+
+/--
+The **signed** variation of `f` on the interval `Icc a b` intersected with the set `s`,
+assuming `f` has locally bounded variation on `s` (so that `to_real` doesn't squash `⊤`).
+-/
+@[nolint unused_arguments]
+noncomputable def variation_from_to (hf : has_locally_bounded_variation_on f s) (a b : α) : real :=
+if a ≤ b then (evariation_on f (s ∩ Icc a b)).to_real else
+            - (evariation_on f (s ∩ Icc b a)).to_real
+
+lemma variation_from_to_self (a : α) : hf.variation_from_to a a = 0 :=
+begin
+  dsimp only [variation_from_to],
+  rw [if_pos (le_refl _), Icc_self, evariation_on.subsingleton, ennreal.zero_to_real],
+  exact λ x hx y hy, hx.2.trans hy.2.symm,
+end
+
+lemma variation_from_to_nonneg_of_le {a b : α} (h : a ≤ b) : 0 ≤ hf.variation_from_to a b :=
+by { dsimp only [variation_from_to], simp only [if_pos h, ennreal.to_real_nonneg], }
+
+lemma variation_from_to_nonpos_of_ge {a b : α} (h : b ≤ a) : hf.variation_from_to a b ≤ 0 :=
+begin
+  dsimp only [variation_from_to],
+  split_ifs,
+  { cases le_antisymm h h_1,
```

Comment:
It's better to use e.g. `split_ifs with h' h'` to specify names rather than using the autogenerated `h_1`.